README.md

```diff
@@ -70,6 +70,86 @@ impl IdAllocator {
 }
 ```
 
+## AddressAllocator
+
+### Design address allocator
+
+This allocator should be used to allocate memory slots, either for memmory-mapped
+I/O or for port I/O or any other kind of address space.
+
+The struct that defined the AddressAllocator contains the ends of the interval
+that is managed and an instance of `interval_tree` as an allocation engine. We
+chose to use `interval_tree` for memory slots allocation because it provides
+better creation performance and more query speed.
+
+```rust
+/// Address allocator representation
+pub struct AddressAllocator {
+    // Begining of the address space that we want to manage.
+    base: u64,
+    // End of the address space that we want to manage.
+    end: u64,
+    // Allocation engine, the allocation logic that is interfaced
```

Comment:
"the ... logic that is interfaced" sounds a bit weird. Maybe just "The Interval Tree allocation engine"?